### PR TITLE
[Workspace] feat: add workspace collaborators page

### PR DESCRIPTION
**Pull request by raintygao**
### Description

feat: add workspace collaborators page

## Screenshot

image

## Testing the changes

Go to workspace collaborator page(http://localhost:6601/w/{workspace_id}/app/collaborators) then see the changes.

## Changelog

- feat: Add workspace collaborators page

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by raintygao** [comment]
The related test cases is still in a progress of being added.

File: src/plugins/workspace/common/constants.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export const WORKSPACE_LIST_APP_ID = 'workspace_list';
 export const WORKSPACE_DETAIL_APP_ID = 'workspace_detail';
 export const WORKSPACE_INITIAL_APP_ID = 'workspace_initial';
 export const WORKSPACE_NAVIGATION_APP_ID = 'workspace_navigation';
+export const WORKSPACE_COLLABORATORS_APP_ID = 'collaborators';
```
Comment (wanglam) [inline src/plugins/workspace/common/constants.ts:17]: Shall we start with `workspace_` prefix like other applications?

File: src/plugins/workspace/public/components/workspace_collaborators/workspace_collaborators.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  if (!currentWorkspace || !isPermissionEnabled) {
+    return null;
+  }
+  return (
+    <EuiPage data-test-subj="workspace-collaborators-panel">
+      <HeaderControl
+        controls={[
+          {
+            description: 'Manage workspace access and permissions.',
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_collaborators/workspace_collaborators.tsx:98]: do we need i18n for description?

File: src/plugins/workspace/public/components/workspace_collaborators/workspace_collaborators.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    currentWorkspace?.permissions ?? {}
+  );
+
+  const isPermissionEnabled = application?.capabilities.workspaces.permissionEnabled;
+
+  const handleSubmitPermissionSettings = async (settings: WorkspacePermissionSetting[]) => {
+    const showErrorNotification = (errorText?: string) => {
+      notifications?.toasts.addDanger({
+        title: i18n.translate('workspace.update.failed', {
+          defaultMessage: 'Failed to update workspace',
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_collaborators/workspace_collaborators.tsx:62]: `Failed to update workspace collaborators`?

File: src/plugins/workspace/public/components/workspace_form/add_collaborator_button.tsx [hunk 4]
```diff
@@ -85,7 +95,18 @@ export const AddCollaboratorButton = ({
...
       ...addedSettings.map(({ collaboratorId, ...rest }) => rest),
     ];
-    await handleSubmitPermissionSettings(newPermissionSettings as WorkspacePermissionSetting[]);
+    const result = await handleSubmitPermissionSettings(
+      newPermissionSettings as WorkspacePermissionSetting[]
+    );
+    if (result?.success) {
+      if (notifications) {
+        notifications.toasts.addSuccess({
+          title: i18n.translate('workspace.workspaceDetail.collaborator.add.success', {
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_form/add_collaborator_button.tsx:104]: nit: should we remove workspaceDetail from the key `workspace.collaborator.add.success`

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx [hunk 5]
```diff
@@ -226,14 +228,27 @@ export const WorkspaceCollaboratorTable = ({
...
           });
-          handleSubmitPermissionSettings(newSettings as WorkspacePermissionSetting[]);
-          setSelection([]);
-          modal.close();
+          const result = await handleSubmitPermissionSettings(
+            newSettings as WorkspacePermissionSetting[]
+          );
+          if (result?.success) {
+            notifications?.toasts?.addSuccess({
+              title: i18n.translate('workspace.detail.collaborator.delete.success', {
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx:241]: nit: same here, should we remove detail for the key

**Comment by Hailong-am** [comment]
image

buttons on both left and right are larger than search bar

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx [hunk 6]
```diff
@@ -309,20 +327,25 @@ export const WorkspaceCollaboratorTable = ({
     {
       field: 'primaryId',
       name: 'ID',
+      width: '30%',
     },
     {
       field: 'displayedType',
       name: 'Type',
```
Comment (Hailong-am) [inline src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx:334]: i18 for name and other columns

**Review by Hailong-am — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx [hunk 7]
```diff
@@ -226,14 +228,27 @@ export const WorkspaceCollaboratorTable = ({
...
           selection.forEach(({ id }) => {
             newSettings = newSettings.filter((_item) => _item.id !== id);
           });
-          handleSubmitPermissionSettings(newSettings as WorkspacePermissionSetting[]);
-          setSelection([]);
-          modal.close();
+          const result = await handleSubmitPermissionSettings(
+            newSettings as WorkspacePermissionSetting[]
+          );
+          if (result?.success) {
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx:239]: Do we need to show error toast if result.success is false?

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**